libraries/Microsoft.Bot.Connector/Authentication/ParameterizedBotFrameworkAuthentication.cs

```diff
@@ -100,7 +100,7 @@ public override async Task<UserTokenClient> CreateUserTokenClientAsync(ClaimsIde
 
             var credentials = await _credentialFactory.CreateCredentialsAsync(appId, _toChannelFromBotOAuthScope, _toChannelFromBotLoginUrl, _validateAuthority, cancellationToken).ConfigureAwait(false);
 
-            return new UserTokenClientImpl(appId, credentials, _oAuthUrl, _httpClient, _logger);
+            return new UserTokenClientImpl(appId, credentials, _oAuthUrl, _httpClientFactory?.CreateClient(), _logger);
```

Comment:
Should this include the _authHttpClient  if the _httpClientFactory is null?
```cs
return new UserTokenClientImpl(appId, credentials, _oAuthUrl, _httpClientFactory?.CreateClient(), _logger);
```

Reply:
No static was just for the JwtTokenExtractor not for calls to ABS - ultimately I'd like to review the usage of this cache.

Inside the UserTokenImpl it creates a new one - and adds the right headers and wires up the Dispose.

Basically if you want efficiency then plug in a httpclientfactory otherwise it still works but less efficient. I don't want to confuse things with the JwtTokenExtractor world.

Reply:
So this test passed when I re-ran. Are we seeing this failure much. If I look at the test it looks problematic - there is a race on the check for connect. The fix I did on Sunday on this same test was to eliminate the null exception (caused by a similar race) anyhow that fix was correcting the basic usage pattern around TaskCompletionSource in the product code anyhow this race looks more like a test problem. I'll have to look it some more.

I'm thinking this is all completely unrelated to this PR incidentally.